lib/internal/streams/pipeline.js

```diff
@@ -178,8 +180,38 @@ function pipeline(...streams) {
 
   let finishCount = 0;
 
+  let hasTransform = [];
+
   function finish(err) {
-    const final = --finishCount === 0;
+    hasTransform = streams.filter((s) => {
```

Comment:
Nit: I think you should use `.some` instead (which would also reduce the:
```js
hasTransform = hasTransform.length > 0
```
Which can confuse other when the variable type changed...)


> **Disclaimer:** not a contributor, just interested in PRs